src/CMakeLists.txt

```diff
@@ -59,6 +59,7 @@ if( OPENSSL_VERSION LESS "1.0.0" )
     #ignore openssl code warnings
     set_source_files_properties( ${openssl_SRCS} PROPERTIES COMPILE_FLAGS ${OPENSSL_IGNORE} )
     list( APPEND digidocpp_SRCS ${openssl_SRCS} )
+    list( APPEND digidocpp_DEFS NO_CMS )
```

Comment:
NO_CMS approach is taken from OpenSSL policy. If feature is not available NO_CMS is defined. Here is workaround with openssl < 1.0.0.

OpenSSL is not buggy. OpenSSL currently only supports PKCS#7 wrapping.
Latvia TimeStamping authority is the only (as far I know) provider who creates CMS wrapping timestamps.